builtin/providers/aws/resource_aws_opsworks_instance.go

```diff
@@ -0,0 +1,1001 @@
+					buf.WriteString(fmt.Sprintf("%s-", m["virtual_name"].(string)))
+					return hashcode.String(buf.String())
+				},
+			},
+
+			"root_block_device": &schema.Schema{
+				// TODO: This is a set because we don't support singleton
+				//       sub-resources today. We'll enforce that the set only ever has
+				//       length zero or one below. When TF gains support for
+				//       sub-resources this can be converted.
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					// "You can only modify the volume size, volume type, and Delete on
+					// Termination flag on the block device mapping entry for the root
+					// device volume." - bit.ly/ec2bdmap
+					Schema: map[string]*schema.Schema{
+						"delete_on_termination": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+							ForceNew: true,
+						},
+
+						"iops": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+
+						"volume_size": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+
+						"volume_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					// there can be only one root device; no need to hash anything
+					return 0
+				},
+			},
+		},
+	}
+}
+
+
+func validateArchitecture(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value != "x86_64" && value != "i386" {
+		errors = append(errors, fmt.Errorf(
+			"%q must be one of \"x86_64\" or \"i386\"", k))
+	}
+	return
+}
+
+func validateAutoScalingType(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value != "load" && value != "timer" {
+		errors = append(errors, fmt.Errorf(
+			"%q must be one of \"load\" or \"timer\"", k))
+	}
+	return
+}
+
+func validateRootDeviceType(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value != "ebs" && value != "instance-store" {
+		errors = append(errors, fmt.Errorf(
+			"%q must be one of \"ebs\" or \"instance-store\"", k))
+	}
+	return
+}
+
+func validateState(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value != "running" && value != "stopped" {
+		errors = append(errors, fmt.Errorf(
+			"%q must be one of \"running\" or \"stopped\"", k))
+	}
+	return
+}
+
+func validateVirtualizationType(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value != "paravirtual" && value != "hvm" {
+		errors = append(errors, fmt.Errorf(
+			"%q must be one of \"paravirtual\" or \"hvm\"", k))
+	}
+	return
+}
+
+func resourceAwsOpsworksInstanceValidate(d *schema.ResourceData) error {
```

Comment:
Are you certain this is the case? I don't see this in the documentation, but haven't had a chance to test myself. Seems odd that this would be restricted since you can add block devices to any other ec2 instance regardless of AMI.